src/lib/GovernorTimelockCompound.sol

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v4.6.0) (governance/extensions/GovernorTimelockCompound.sol)
+
+pragma solidity ^0.8.0;
+
+import {Governor} from "flexible-voting/src/GovernorCountingFractional.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {IGovernorTimelock} from
+  "@openzeppelin/contracts/governance/extensions/IGovernorTimelock.sol";
+import {ITimelockAlpha} from "src/interfaces/ITimelockAlpha.sol";
+import {Timers} from "@openzeppelin/contracts/utils/Timers.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {IERC165} from "@openzeppelin/contracts/interfaces/IERC165.sol";
+import {IGovernor} from "@openzeppelin/contracts/governance/IGovernor.sol";
+
+/**
+ * @dev Extension of {Governor} that binds the execution process to a Compound Timelock. This adds a
+ * delay, enforced by
+ * the external timelock to all successful proposal (in addition to the voting duration). The
+ * {Governor} needs to be
+ * the admin of the timelock for any operation to be performed. A public, unrestricted,
+ * {GovernorTimelockCompound-__acceptAdmin} is available to accept ownership of the timelock.
+ *
+ * Using this model means the proposal will be operated by the {TimelockController} and not by the
+ * {Governor}. Thus,
+ * the assets and permissions must be attached to the {TimelockController}. Any asset sent to the
+ * {Governor} will be
+ * inaccessible.
+ *
+ * _Available since v4.3._
+ */
+abstract contract GovernorTimelockCompound is IGovernorTimelock, Governor {
+  using SafeCast for uint256;
+  using Timers for Timers.Timestamp;
+
+  struct ProposalTimelock {
+    Timers.Timestamp timer;
+  }
+
+  // This was changed from the original openzeppelin contract to work with the pooltogether timelock
```

Comment:
For all of these comments, can we expand them to include details of what they were changed from in the original source. A permalink to the OZ repo to the line in question would also be good.

Sorry to be annoying about documenting this, but it's definitely a sensitive area and something someone will definitely have to take into account in the future, so we want to be responsible and leave plenty of breadcrumbs.

Reply:
No problem, let me know if even more context is necessary! Happy to add